### PR TITLE
ts strict - getAttribute

### PR DESCRIPTION
**Pull request by tedpatrick**
Working through TS strict errors related to getAttribute.

Element.getAttribute returns string | null

All existing code ignores the potential null and assumes string.

1. Add a util for getAttribute which both tests for hasAttribute and supports string, null
2. Implement across pyscript

### CODE REVIEW

File: pyscriptjs/src/utils.ts [hunk 1]
```diff
@@ -1,46 +1,44 @@
...
-function removeClasses(element: HTMLElement, classes: Array<string>) {
+export function removeClasses(element: HTMLElement, classes: string[]) {
     for (const entry of classes) {
         element.classList.remove(entry);
     }
 }
 
-function getLastPath(str: string): string {
-    return str.split('\\').pop().split('/').pop();
+export function getLastPath(str: string): string | undefined {
```
Comment (madhur-tandon) [inline pyscriptjs/src/utils.ts:13]: Why a possibility of `undefined` here?
I ask since -- `str.split('\\').pop()` will always return a string, even if the original string doesn't have any `\\` in it i.e. `"hh".split('\\').pop() == "hh"`

Comment (tedpatrick) [reply pyscriptjs/src/utils.ts:13]: We will get to that. This PR focuses on types affecting getAttribute only.

Comment (tedpatrick) [reply pyscriptjs/src/utils.ts:13]: looking again...

Comment (tedpatrick) [reply pyscriptjs/src/utils.ts:13]: Oh. the ?. in the next line

```ts
return str.split('\\').pop()?.split('/').pop();
```

Comment (madhur-tandon) [reply pyscriptjs/src/utils.ts:13]: Yeah, I mean, why do we need the `?`

Comment (tedpatrick) [reply pyscriptjs/src/utils.ts]: fixed

File: pyscriptjs/src/utils.ts [hunk 2]
```diff
@@ -108,6 +106,12 @@ function globalExport(name: string, obj: any) {
     _global[name] = obj;
 }
 
-
-
-export { addClasses, removeClasses, getLastPath, ltrim, htmlDecode, guidGenerator, showError, handleFetchError, readTextFromPath, inJest, globalExport, };
+export function getAttribute(el:Element, attr:string ):string | null {
```
Comment (madhur-tandon) [inline pyscriptjs/src/utils.ts:109]: I guess a linter could fix issues with spacing etc. such as `(el:Element, attr:string )` --> `(el:Element, attr:string)` but it's minor.

But this kind of stuff might be present in other places.

**Review by madhur-tandon — APPROVED** [review]

File: pyscriptjs/src/components/pyrepl.ts [hunk 3]
```diff
@@ -25,7 +25,10 @@ export function make_PyRepl(runtime: Runtime) {
 
     let initialTheme: string;
     function getEditorTheme(el: BaseEvalElement): string {
-        return initialTheme || (initialTheme = el.getAttribute('theme'));
+        if( !initialTheme ){
+            initialTheme = el.getAttribute('theme') || "";
```
[the remark this answers is not in the document]
Comment (tedpatrick) [reply pyscriptjs/src/components/pyrepl.ts]: Added
